kytos/utils/decorators.py

```diff
@@ -65,6 +65,21 @@ def authenticate(self):
         username = self.config.get('auth', 'user')
         password = getpass("Enter the password for {}: ".format(username))
         response = requests.get(endpoint, auth=(username, password))
+        if response.status_code == 401:
+            invTokenStr = "{\"error\":\"Token not sent or expired: " \
+                          "Signature has expired\"}\n"
+            # pylint: disable=superfluous-parens
+            if (invTokenStr == response.content.decode()):
+                print("Seems the token was not set or is expired!"
+                      "Please run \"kytos napps upload\" again.")
+                LOG.error(response.content)
+                LOG.error('ERROR: %s: %s', response.status_code,
+                          response.reason)
+                print("Press Ctrl+C or CTRL+Z to stop the process.")
+                user = input("Enter the username: ")
+                self.config.set('auth', 'user', user)
+                self.authenticate()
```

Comment:
I think this recursive call can raise an error when the recursion ends, because the following check (line 83) will be executed even if the user has already authenticated.

Reply:
Maybe all instructions within this 'if' can be moved to inside the: `if response.status_code != 201`.

Reply:
I applied your suggested changes, and tested everything to make sure it works (it does), however, there is an OSError causing the decorators.py unit test to fail. This makes no sense. Is there a way to ignore this error? It seems unnecessary.